### PR TITLE
migrate single server to compose

### PR DESCRIPTION
**Pull request by davejrt**
A script to migrate Sourcegraph single server deployments to compose:

I chose editing the dump file for sql commands as the bash foo to escape quotes was awful. Happy to revisit this if someone disagrees but I feel as though I burnt too much time here before it occurred to me it'd just be easier do it this way. 

fixes https://github.com/sourcegraph/sourcegraph/issues/11600

### CODE REVIEW

File: tools/migrate.sh [hunk 1]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Please ensure the variales in migration vars have been assigned before running this script.
```
Comment (uwedeportivo) [inline tools/migrate.sh:3]:
```suggestion
# Please ensure the variables in migration vars have been assigned before running this script. 
```

File: tools/migrate.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Please ensure the variales in migration vars have been assigned before running this script. 
+
+set -euxo pipefail
+
+# shellcheck source=migration_vars
+source migration_vars
```
Comment (uwedeportivo) [inline tools/migrate.sh:8]: maybe check here that the variables you assume as set are actually set

**Review by uwedeportivo — APPROVED** [review]

File: tools/migrate.sh (hunk 2 above)
Comment (davejrt) [reply tools/migrate.sh:8]: OK yeah, in testing I never bothered to check for this and assumed `-u` would pick that up but it doesn't work if you're sourcing the file.

Comment (davejrt) [reply tools/migrate.sh:9]: Running the `source <file>` in a subshell fixes this though. New commit up now

File: tools/migrate.sh [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+echo "Obtaing sourcegraph container ID"
+container_id=$(ssh "${src_host}" docker ps | grep "${image}:${version}" | cut -f 1 -d ' ')
+
+# Dump sourcegraph db and alter script for migration
+echo "Dumping sourcegraph database"
+ssh ${src_host} "docker exec ${container_id} pg_dump -C --username=postgres sourcegraph" >${sg_dump_file}
+sed -i '/^CREATE DATABASE/a CREATE USER postgres WITH SUPERUSER CREATEROLE CREATEDB REPLICATION BYPASSRLS;' ${sg_dump_file} >/dev/null 2>&1
+printf '\connect -reuse-previous=on dbname=postgres\nDROP DATABASE sg;\nALTER DATABASE "sourcegraph" RENAME TO sg;\nALTER DATABASE sg OWNER TO sg;\n' >>${sg_dump_file}
+
+# Dump codeintel db and alter script for migration
```
Comment (daxmc99) [inline tools/migrate.sh:28]: cc @slimsag, is codeinsights supported in our `server` image?
Not sure if we need to add migration steps here as well

Comment (slimsag) [reply tools/migrate.sh:28]: It is not supported. Deploying TimescaleDB inside of single-container would take a fair amount of effort, and given it is for demo purposes only we have made the decision for now to not support it. (a more "minimal" version, without any TimescaleDB backend, might work - though)

**Pull request merged**